pipeline: parsers: decoders: general cleanup (#1878)
fluent/fluent-bit-docs: pipeline/parsers/decoders.md

# Decoders

> There are cases where the log messages that you want to parse contain encoded data. A typical use case can be found in containerized environments with Docker. Docker logs its data in JSON format, which uses escaped strings.

Reviewer: ```suggestion
There are cases where the log messages you want to parse contain encoded data. A typical use case can be found in containerized environments with Docker. Docker logs its data in JSON format, which uses escaped strings.
```

{"log":"{\"status\": \"up and running\"}\r\n","stream":"stdout","time":"2018-03-09T01:01:44.851160855Z"}
```

> The original message is handled as an escaped string. Fluent Bit wants to use the original structured message and not a string.

Reviewer: Wanting something is what a human would do, and not an application. It can request things. It can do things. But it doesn't want things.

```suggestion
The original message is handled as an escaped string. Fluent Bit will use the original structured message, and not a string.
```

## Get started

> Decoders are a built-in feature of parsers in Fluent Bit. Each parser definition can optionally set one or more decoders. There are two types of decoders:

Reviewer: You don't have to use this exact suggestion, but I'm trying to get us out of the habit of saying "here are the two types" or "select from these three options", because putting a number in there makes potential technical debt. What if they add another type? Then we need to not only add the info, but to update the intro section for that list. It's a small thing, but it adds up.

```suggestion
Decoders are a built-in feature of parsers in Fluent Bit. Each parser definition can optionally set one or more decoders. Select from one of these decoder types:
```

> Decoders are a built-in feature available through the `parsers.conf` file. Each parser definition can optionally set one or more decoders. There are two types of decoders:

Author: ```suggestion
Decoders are a built-in feature of parsers in Fluent Bit. Each parser definition can optionally set one or more decoders. There are two types of decoders:
```
